lib/rbs/test/type_check.rb

```diff
@@ -331,7 +331,7 @@ def value(val, type)
         when Types::Variable
           true
         when Types::Literal
-          type.literal == val
+          defined?(val.==) and type.literal == val
```

Comment:
This is needed for the `Kernel#test`, as the `int` variants of the "catch all" variants don't define `==`.